### PR TITLE
feat: add pagination for list appeal and list grant

### PR DESCRIPTION
**Pull request by lifosmin**
proton PR: https://github.com/goto/proton/pull/46

### CODE REVIEW

File: buf.gen.yaml [hunk 1]
```diff
@@ -5,15 +5,15 @@ plugins:
...
     out: api/proto
     opt: paths=source_relative
   - plugin: buf.build/grpc-ecosystem/openapiv2:v2.15.1
     out: third_party/OpenAPI
   # See https://github.com/bufbuild/protoc-gen-validate/issues/523
-  - remote: buf.build/jirkad/plugins/protoc-gen-validate:v0.6.7
-    out: api/proto
-    opt:
-      - paths=source_relative
-      - lang=go
```
Comment (rahmatrhd) [inline buf.gen.yaml:19]: let's revert this file to the original

File: api/handler/v1beta1/grant.go [hunk 2]
```diff
@@ -26,6 +26,8 @@ func (s *GRPCServer) ListGrants(ctx context.Context, req *guardianv1beta1.ListGr
 		CreatedBy:     req.GetCreatedBy(),
 		Owner:         req.GetOwner(),
 		OrderBy:       req.GetOrderBy(),
+		Size:          int(req.GetSize()),
+		Offset:        int(req.GetOffset()),
 	}
```
Comment (rahmatrhd) [inline api/handler/v1beta1/grant.go:31]: I think we're missing on ListUserGrants

**Review by rahmatrhd — CHANGES_REQUESTED** [review]

File: internal/store/postgres/appeal_repository_test.go [hunk 3]
```diff
@@ -290,6 +290,25 @@ func (s *AppealRepositoryTestSuite) TestFind() {
...
+	s.Run("Should return an array size and offset of n on success", func() {
+		appeals, err := s.repository.Find(context.Background(), &domain.ListAppealsFilter{
+			Size:   1,
+			Offset: 1,
+		})
+
+		s.NoError(err)
+		s.Len(appeals, 1)
+	})
 }
```
Comment (bsushmith) [inline internal/store/postgres/appeal_repository_test.go:312]: Can you rename the test cases or make them table tests if needed?

File: internal/store/postgres/grant_repository_test.go [hunk 4]
```diff
@@ -179,6 +179,23 @@ func (s *GrantRepositoryTestSuite) TestList() {
...
+		appeals, err := s.repository.List(context.Background(), domain.ListGrantsFilter{
+			Size: 1,
+		})
+
+		s.NoError(err)
+		s.Len(appeals, 1)
+	})
+	s.Run("Should return an array size and offset of n on success", func() {
+		appeals, err := s.repository.List(context.Background(), domain.ListGrantsFilter{
+			Size:   1,
```
Comment (bsushmith) [inline internal/store/postgres/grant_repository_test.go:192]: Can you rename the test cases or make them table tests if needed?

**Review by rahmatrhd — APPROVED** [review]
LGTM

**Pull request merged**